peachjam/adapters/adapters.py

```diff
@@ -345,6 +346,37 @@ def update_document(self, url):
 
         self.set_parent(document, created_doc)
         self.fetch_relationships(document, created_doc)
+        self.download_and_save_document_images(document, created_doc)
+
+    def list_images_from_content_api(self, document):
+        links = document["links"]
+        if links:
+            for link in links:
+                if link["href"].endswith("media.json"):
+                    response = self.client_get(link["href"])
+                    if response.json()["results"]:
+                        return response.json()["results"]
+
+    def download_and_save_document_images(self, document, created_document):
+        image_list = self.list_images_from_content_api(document)
+        if image_list:
+            for result in image_list:
+                if result["mime_type"].startswith("image/"):
+                    with NamedTemporaryFile() as file:
+                        r = self.client_get(result["url"])
+                        file.write(r.content)
+
+                        Image.objects.get_or_create(
```

Comment:
Knowing when to update the file is difficult. I suggest simply deleting all images associated with the file right at the start of the function (line 360). Then assume that the filenames are unique, and simply create the file rather than `get_or_create`.